### ISSUE TITLE
non_maximum_suppression consuming memory

### ISSUE THREAD

**Issue by tesaho** [eriklindernoren/PyTorch-YOLOv3#187]
I've been running a validation set of 4 images from COCO and my laptop seems to consume memory in the while loop of the non_maximum_suppression function after the first few epochs.  Is there a way to release the memory?  Has anyone encountered this?  I am using a batchsize of 1 on ubuntu 18.04, with 16 gb ram, gtx 1070, cuda 10.  Training seems to leave my memory around 8gb with other programs running.

**Comment by jwkvam** [comment]
I'm noticing the same thing, it gets stuck in this loop https://github.com/eriklindernoren/PyTorch-YOLOv3/blob/master/utils/utils.py#L251. The `detections.size(0)` is not decreasing whereas it did in the previous epochs.

**Comment by jwkvam** [comment]
Try decreasing the learning rate. I found some `nan`s and it seems to be working so far. Looks like the default is 1e-3.

https://github.com/eriklindernoren/PyTorch-YOLOv3/blob/master/train.py#L77
https://pytorch.org/docs/stable/optim.html#torch.optim.Adam

**Comment by tesaho** [comment]
Thanks! This worked for me.
